src/deadline_worker_agent/sessions/actions/run_attachment_download.py

```diff
@@ -275,8 +275,10 @@ def start(
         manifest_paths_by_root = session._asset_sync._check_and_write_local_manifests(
             merged_manifests_by_root=merged_manifests_by_root,
             manifest_write_dir=str(session.working_directory),
+            manifest_name_suffix="step" if self._step_details else "job",
         )
-        session.manifest_paths_by_root = manifest_paths_by_root
+        # Set the manifests by root mapping to session for attachment upload to determine output
+        [session.add_manifest_path(root=r, path=m) for r, m in manifest_paths_by_root.items()]
```

Comment:
nit: the list comprehension is clever trick to keep this code on a single line, but it makes it slightly less apparent on first glance that this is a loop with side-effects. How do you feel about switching to a regular for loop:

```suggestion
        for root_name, root_path in manifest_paths_by_root.items():
            session.add_manifest_path(root=root_name, path=root_path)
```

Reply:
I found the list comprehension to be concise, but I don't mind switching to regular loop.